upf/test/test_pddl_planner.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2021 AIPlan4EU project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import upf
+from upf.environment import get_env
+from upf.shortcuts import *
+from upf.test import TestCase, main
+from upf.pddl_solver import PDDLSolver
+
+
+FILE_PATH = os.path.dirname(os.path.abspath(__file__))
+
+
+class ENHSP(PDDLSolver):
+    def __init__(self):
+        PDDLSolver.__init__(self, False)
+
+    def _get_cmd(self, domanin_filename: str, problem_filename: str, plan_filename: str) -> List[str]:
+        return ['java', '-jar', os.path.join(FILE_PATH, '..', '..', '.planners', 'enhsp-20', 'enhsp.jar'),
+                '-o', domanin_filename, '-f', problem_filename, '-sp', plan_filename]
+
+
+class TestPDDLPlanner(TestCase):
+    def setUp(self):
+        TestCase.setUp(self)
+        env = get_env()
+        if not os.path.isfile(os.path.join(FILE_PATH, '..', '..', '.planners', 'enhsp-20', 'enhsp.jar')):
+            self.skipTest('ENHSP not found!')
+        env.factory.add_solver('enhsp', 'upf.test.test_pddl_planner', 'ENHSP')
+
+    def test_basic(self):
+        x = upf.Fluent('x')
+        a = upf.Action('a')
+        a.add_precondition(Not(x))
+        a.add_effect(x, True)
+        problem = upf.Problem('basic')
+        problem.add_fluent(x)
+        problem.add_action(a)
+        problem.set_initial_value(x, False)
+        problem.add_goal(x)
+
+        planner = OneshotPlanner(name='enhsp')
+
+        plan = planner.solve(problem)
+        self.assertEqual(len(plan.actions()), 1)
+        self.assertEqual(plan.actions()[0].action(), a)
+        self.assertEqual(len(plan.actions()[0].parameters()), 0)
```

Comment:
Maybe you can add a `test_examples` that iterates through all the example problems and try to solve them with ENHSP?